### PR TITLE
Add i.MX 8ULP Machines

### PR DESCRIPTION
**Pull request by angolini**
I have only built the source code. I don't have the board so I cannot test it. In case anyone is available to test, please, let me know the result.

It's important to note that the `IMXBOOT_TARGETS` tries to use dualboot, but the `m33` binaries are not included:

```
IMXBOOT_TARGETS_SD   = "flash_singleboot flash_dualboot"
IMXBOOT_TARGETS_FSPI = "flash_dualboot_flexspi"
IMXBOOT_TARGETS_ND   = ""
```

### CODE REVIEW

**Comment by thochstein** [comment]
+@junzhuimx 

Thanks, @angolini, but unfortunately 8ULP is not in production yet. Our website does show a single board:

https://www.nxp.com/products/power-management/pmics-and-sbcs/pmics/pca9460-evaluation-board:KIT-PCA9460-EVB

The name is different. Maybe we should hold off on this?

File: conf/machine/imx8ulp-9x9-lpddr4-evk.conf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#@TYPE: Machine
+#@NAME: i.MX 8ULP 9x9 EVK
+#@SOC: i.MX8ULP
+#@DESCRIPTION: Machine configuration for NXP i.MX 8ULP 9x9 Evaluation Kit with LPDDR4
```
Comment (thochstein) [inline conf/machine/imx8ulp-9x9-lpddr4-evk.conf:4]: These boards with `9x9` in the name seem to be only for internal use, so not upstreamed.

File: conf/machine/imx8ulpevk.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#@TYPE: Machine
+#@NAME: i.MX 8ULP EVK
+#@SOC: i.MX8ULP
+#@DESCRIPTION: Machine configuration for NXP i.MX 8ULP Evaluation Kit
```
Comment (thochstein) [inline conf/machine/imx8ulpevk.conf:4]: Don't include this file either, we still haven't fully adapted since the machine renames.

Comment (angolini) [reply conf/machine/imx8ulpevk.conf:4]: Thanks @thochstein , let me rework the patchset.

Comment (angolini) [reply conf/machine/imx8ulpevk.conf]: @thochstein done!

**Comment by angolini** [comment]
@thochstein I still don't have access to the board, but in the future, we are going to have it. It would be good to test with the reviewed patchset. Would you mind reviewing the PR?

File: conf/machine/imx8ulp-lpddr4-evk.conf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+KERNEL_DEVICETREE += " \
+    freescale/imx8ulp-evk-epdc.dtb \
+    freescale/imx8ulp-evk-flexio-i2c.dtb \
+    freescale/imx8ulp-evk-i3c.dtb \
+    freescale/imx8ulp-evk-lpa.dtb \
+    freescale/imx8ulp-evk-lpspi-slave.dtb \
+    freescale/imx8ulp-evk-nd.dtb \
+    freescale/imx8ulp-evk-rk055hdmipi4m.dtb \
+    freescale/imx8ulp-evk-rk055hdmipi4mv2.dtb \
+    freescale/imx8ulp-evk-sof-btsco.dtb \
```
Comment (thochstein) [inline conf/machine/imx8ulp-lpddr4-evk.conf:20]: Use `${KERNEL_DEVICETREE_BASENAME}` here.

File: conf/machine/include/imx8ulp-evk.inc [hunk 4]
```diff
@@ -0,0 +1,44 @@
+MACHINEOVERRIDES =. "mx8:mx8ulp:"
```
Comment (thochstein) [inline conf/machine/include/imx8ulp-evk.inc:1]: I believe this should be just `MACHINEOVERRIDES =. "mx8ulp"`.

That said, when I grep I see the same pattern for a few more SOCs.

```
conf/machine/imx7dsabresd.conf:MACHINEOVERRIDES =. "mx7:mx7d:"
conf/machine/imx7ulpevk.conf:MACHINEOVERRIDES =. "mx7:mx7ulp:"
conf/machine/imx8qm-mek.conf:MACHINEOVERRIDES =. "mx8:mx8qm:"
```

@otavio, can you please confirm that `mx8` is not needed here?

File: conf/machine/imx8ulp-lpddr4-evk.conf (hunk 3 above)
Comment (thochstein) [reply conf/machine/imx8ulp-lpddr4-evk.conf]: Thanks, @angolini, but there are two more parts, using `append:use-nxp-bsp`, and moving to `.inc`. Actually, leaving it here is fine for now. We can move anything common to `.inc` if/when we add `ddr4` version.

File: recipes-bsp/firmware-sentinel/firmware-sentinel_0.8.bb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+inherit fsl-eula-unpack use-imx-security-controller-firmware deploy
+
+SRC_URI = "${FSL_MIRROR}/${BP}.bin;fsl-eula=true"
+SRC_URI[md5sum] = "be47a5e59c1192ee36246af97d5d1532"
+SRC_URI[sha256sum] = "1003d4c6773c153ea341911a74e25c249423644f70f3d8f8d085599e00770b3f"
+
+do_compile[noexec] = "1"
+
+do_install() {
```
Comment (otavio) [inline recipes-bsp/firmware-sentinel/firmware-sentinel_0.8.bb:16]:
```
do_install[noexec] = "1"
```

Comment (otavio) [reply recipes-bsp/firmware-sentinel/firmware-sentinel_0.8.bb:16]: Please check if the nopackages class might be used.

File: conf/machine/include/imx8ulp-evk.inc (hunk 4 above)
Comment (otavio) [reply conf/machine/include/imx8ulp-evk.inc:1]: It is. It is meant to reuse the `mx8` common override.

**Review by otavio — CHANGES_REQUESTED** [review]

File: conf/machine/include/imx8ulp-evk.inc (hunk 4 above)
Comment (thochstein) [reply conf/machine/include/imx8ulp-evk.inc:1]: If that is the case, then it seems surprising we don't see the pattern more.

**Comment by angolini** [comment]
@otavio @thochstein please let me know if there is anything else I'm missing, because I think there is nothing else from my side so far

**Review by otavio — APPROVED** [review]

File: conf/machine/include/imx8ulp-evk.inc (hunk 4 above)
Comment (thochstein) [reply conf/machine/include/imx8ulp-evk.inc:1]: Isn't `mx8` unused since we changed all the overrides to `mx8-*-bsp`.

**Pull request merged**